src/cswinrt/code_writers.h

```diff
@@ -7755,25 +7761,18 @@ NativeMemory.Free((void*)abiToProjectionVftablePtr);
             // check for !authored type
             !(settings.component && settings.filter.includes(type)))
         {
-            bool hasOverridableAttribute = false;
+            bool write_vftable_ptr = false;
             auto exclusive_to_type = get_exclusive_to_type(type);
-            for (auto&& iface : exclusive_to_type.InterfaceImpl())
+            for (auto&& [interface_name, attributed_type] : get_attributed_types(w, exclusive_to_type))
             {
-                for_typedef(w, get_type_semantics(iface.Interface()), [&](auto interface_type)
-                {
-                    if (type == interface_type && is_overridable(iface))
-                    {
-                        hasOverridableAttribute = true;
-                    }
-                });
-
-                if (hasOverridableAttribute)
+                if ((attributed_type.activatable || attributed_type.statics || attributed_type.overridable) && attributed_type.type == type)
```

Comment:
Do we want to start generating the vftbl for these scenarios?  In most cases other than I guess for this test case, they are not used.  I wonder if there is another way to handle scenarios like the test case.

Reply:
I think it makes sense if the interface is public.